lib/src/map/options/enabled_gestures.dart

```diff
@@ -0,0 +1,254 @@
+import 'package:flutter_map/flutter_map.dart';
+import 'package:meta/meta.dart';
+
+@immutable
+class EnabledGestures {
+  /// Private constructor, use the constructors [InteractiveFlags.all] or
+  /// [InteractiveFlags.none] instead to enable or disable all gestures by default.
+  /// If you want to define your enabled gestures using bitfield operations,
+  /// use [InteractiveFlags.bitfield] instead.
+  const EnabledGestures._({
+    required this.drag,
+    required this.flingAnimation,
+    required this.twoFingerMove,
+    required this.twoFingerZoom,
+    required this.doubleTapZoomIn,
+    required this.doubleTapDragZoom,
+    required this.scrollWheelZoom,
+    required this.twoFingerRotate,
+    required this.ctrlDragRotate,
+  });
+
+  /// Shortcut constructor to allow all gestures that don't rotate the map.
+  const EnabledGestures.noRotation()
```

Comment:
This could be an indicator that we need more categorization of these gestures, so larger swathes can be toggled at once. For example, movement. Again, maybe enums, maybe mixins, maybe inheritance, maybe just all possible constructors - not sure.

Reply:
Is there really a meaningful amount of users that need to toggle all movement or all zooming in a group that it makes it worth the additional shortcut code for every variant? No rotation is a common use case, clearly written in the docs and still gets asked from time to time therefore a shortcut makes sense in my opinion. For the other gesture groups however it's not much more work to toggle those flags themselfs. `EnabledGestures` provides users with a gesture whitelist and blacklist approach that is in my opinion sufficient. 
Could you provide my with an use case for extended grouping of gestures?

Reply:
While wrtiting https://github.com/fleaflet/flutter_map/pull/1733#discussion_r1429228899 an idea came in my mind that would allow this functionality in a really easy way. Really don't know why this hasn't come into my mind earlier:
```dart
const EnabledGestures.fromGroups({
  required bool move,
  required bool zoom,
  required bool rotate,
}) : this._(
        drag: move,
        twoFingerMove: move,
        flingAnimation: move,
        doubleTapDragZoom: zoom,
        doubleTapZoomIn: zoom,
        scrollWheelZoom: zoom,
        twoFingerZoom: zoom,
        twoFingerRotate: rotate,
        ctrlDragRotate: rotate,
      );
```

In addition, there could be another wither, something like `copyWithGroups()` and that's it.

Reply:
Yeah, I do like that. Maybe a better name would be with `byGroup`?

Reply:
Sounds good, renamed the constructor.